hadoop-hdfs-project/hadoop-hdfs/src/site/markdown/HdfsBlockPlacementPolicies.md

```diff
@@ -0,0 +1,98 @@
+<!---
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License. See accompanying LICENSE file.
+-->
+
+BlockPlacementPolicies
+======================
+* [Block Placement Policies](#Block_Placement_Policies)
+	* [Introduction](#Introduction)
+	* [Additional Types of Policies](#Policy_Types)
+		* [BlockPlacementPolicyRackFaultTolerant](#Block_PlacementPolicy_RackFaultTolerant)
+		* [BlockPlacementPolicyWithNodeGroup](#BlockPlacementPolicy_With_NodeGroup)
+		* [BlockPlacementPolicyWithUpgradeDomain](#BlockPlacementPolicy_With_UpgradeDomain)
+
+##Introduction
+By default HDFS supports BlockPlacementPolicyDefault. Where one block on local and copy on 2 different nodes of same remote rack. Additional to this HDFS supports 3 different pluggable block placement policies. Users can choose the policy based on their infrastructre and use case. This document describes the detailed information about the type of policies with its use cases and configuration.
+
+
+### BlockPlacementPolicyRackFaultTolerant
+
+BlockPlacementPolicyRackFaultTolerant can be used to split the placement of blocks across multiple rack.By default with replication of 3 BlockPlacementPolicyDefault will  put one replica on the local machine if the writer is on a datanode, otherwise on a random datanode in the same rack as that of the writer, another replica on a node in a different (remote) rack, and the last on a different node in the same remote rack. So totally 2 racks will be used, in sceneraio like 2 racks going down at the same time will cause data inavailability where using BlockPlacementPolicyRackFaultTolerant will helop in placing 3 blocks on 3 different racks.
+https://issues.apache.org/jira/browse/HDFS-7891
```

Comment:
This is meant for Erasure Coded blocks, not replicated blocks.
In fact, this is the default block placement policy for EC blocks.
I don't think any one used it for replicated blocks.

Reply:
@jojochuang  
Scenario like 2 racks going down or multiple nodes across 2 racks going down may cause missing blocks.
so we have described about the BlockPlacementPolicyRackFaultTolerant which can be used to replicate the blocks, not on same rack.